On x86_64 host with:
pythondir: /usr/lib/python2.6/site-packages/gdb
libdir:    /usr/lib64

prefix got set to "/usr/lib" but the code expects a trailing "/" (missing then
one "../" path component).

--- ./libstdc++-v3-python-r151798/hook.in-orig	2009-06-18 21:12:37.000000000 +0200
+++ ./libstdc++-v3-python-r151798/hook.in	2009-09-17 23:37:35.000000000 +0200
@@ -40,7 +40,7 @@ if gdb.current_objfile () is not None:
     # In some bizarre configuration we might have found a match in the
     # middle of a directory name.
     if prefix[-1] != '/':
-        prefix = os.path.dirname (prefix)
+        prefix = os.path.dirname (prefix) + '/'
 
     # Strip off the prefix.
     pythondir = pythondir[len (prefix):]
